CMakeLists.txt

```diff
@@ -5,6 +5,9 @@ cmake_minimum_required(VERSION 3.10)
 # but semantic versioning should be considered
 project(mpl VERSION 0.4.0 LANGUAGES CXX C)
 
+# https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html
+string(COMPARE EQUAL ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_SOURCE_DIR} PROJECT_IS_TOP_LEVEL)
+
```

Comment:
To my understanding, `PROJECT_IS_TOP_LEVEL` is built-in feature of CMake 3.21 and later. Instead of reinventing the wheel, one should just raise the `cmake_minimum_required`.